### PR TITLE
refactor login logging and example

### PR DESCRIPTION
**Pull request by steveeJ**

### CODE REVIEW

File: src/v2/auth.rs [hunk 1]
```diff
@@ -130,8 +130,24 @@ impl Client {
...
+            .and_then(|token_auth: TokenAuth| {
+                let mut t = token_auth.token().to_string();
+
+                if t == "unauthenticated" {
+                    bail!("received token with value '{}'", t)
+                } else if t.is_empty() {
+                    bail!("received an empty token")
+                };
+
+                // mask the token before logging it
```
Comment (lucab) [inline src/v2/auth.rs]: If we like minimalism, I'm fine with just logging the token length here instead.

File: src/v2/auth.rs [hunk 2]
```diff
@@ -151,22 +167,15 @@ impl Client {
...
+        let req = self.build_reqwest(reqwest::async::Client::new().get(url.clone()));
+        let req = if let Some(t) = token {
+            req.bearer_auth(t)
         } else {
             debug!("is_auth called without token");
+            req
         };
 
-        trace!("Sending reqwest '{:?}'", req);
+        trace!("Sending reqwest to '{}'", url);
```
Comment (lucab) [inline src/v2/auth.rs]: s/reqwest/request/ I guess?

Comment (steveeJ) [reply src/v2/auth.rs]: Aren't they synonymous now in the context of Rust programs? :wink:

File: src/v2/auth.rs (hunk 1 above)
Comment (steveeJ) [reply src/v2/auth.rs]: If you're not opposed to the masked string I'd opt for leaving it in. Please make the call

Comment (lucab) [reply src/v2/auth.rs]: Code is already there, let's keep it.

**Pull request merged**